Menus/MainMenu/MainMenu.cs

```diff
@@ -121,6 +109,25 @@ void Start()
 			UpdateToolButtons();
 		}
 
+		public void SetVisible(bool visible, bool temporary = false)
+		{
+			if (m_Visible != visible)
+			{
+				m_Visible = visible;
+				if (m_MainMenuUI)
+				{
+					m_MainMenuUI.visible = visible;
+					if (!temporary)
+						SendVisibilityPulse();
+				}
+			}
+		}
+
+		public bool GetVisible()
```

Comment:
Oh, I see what you mean. Hm... that still breaks the "single bool property" convention, unless we go to an enum for all UI visibilities. Would it make sense to just formalize MenuHideFlags and set that on the UI objects?

Maybe we should just have a chat about this tomorrow.